### DISCUSSION TITLE
Winge Macronizer

### DISCUSSION THREAD

**Discussion by nkprasad12** [nkprasad12/morcus-net#47] [Paper Discussion]
@scarpart I think this might be a good format for notes.
https://cl.lingfil.uu.se/exarb/arch/winge2015.pdf

**Comment by scarpart** [comment]
Much better, yeah. I've also taken the liberty to create discussions (after this one) about the specific tools mainly so that I can remind myself of their use without having to go back to the paper.

## Things to keep in mind
1. Any methods that would benefit from manual correction? Anything whatsoever that would allow us to use manual correction? This would be hugely valuable.
2. Any ways to speed up the CLTK pipeline?

That said, onto the paper.

**Comment by scarpart** [comment]
# Automatic Annotation of Latin Vowel Length, Johan Winge (Alatius), 2015.

## TL;DR 
TODO

## 1. On Introductory Topics

**(1.1)** There are extensive Laitn corpora with manually annotated morphosynctactic information primarily in the form of dependency treebanks. Three are available freely:
- The Latin Dependency Treebank (LDT)
- The Latin part of the PROIEL treebank 
- The Index Thomisticus Treebank (IT-TB)

**(1.2)** _HunPos_ parses the text by assigning a sequence of tags to a sequence of tokens, using probabilities of a sequence appearing in the training phase. Given a sequence of tokens, the most likely sequence of tags is generated. 

It operates under the assumption that the probability of the tags depends on the surrounding tags and that the probabilities are all independent.

It calculates the result as a result of two probabilities: (1) the probability of seeing the word given the tag (out of all instances of the tag _t_ in the training data, how many times is it paired with the word _w_) and (2) the probability of seeing the tag given the preceding tags. 

**Important:** _HunPos_ adds the possibility to use the information in an external morphological lexicon., HunPos will then restrict the set of possible tags for an unseen word to those given in the lexicon. The tags are then weighted as before, using statistics from words with similar suffix.

**(1.3)** _RFTagger_ is meant to mitigate the problems of sparse data no large tag sets. Larger amounts of possible _n-grams_ implicates in the reduce likelyhood of correctly tagging new data. With this in mind, the tags are decomposed into feature vectors, instead of being treated as atomic units.

An interesting thing is that instead of conditioning the morphological attribute probabilities on the closest preceding tags, the most relevant combination of features are chosen by the creation of decision trees. I.e. to check if a tag is `ablative`, check if the preceding one is `preposition`, and so on. This is done in a separate Binary Decision Tree.

**NOTE TO SELF**: I had an idea here, but I think I accidentally deleted this part from my notes. I can probably find the part where I was, though, and remember it. 

**(1.4)** Latin's free word order and rich morphology make it so it's possible for a word to agree with another, distant one. This makes it hard for parsing and tagging in reduced contexts. The syntax and morphology are complementory, therefore, according to Lee et al. (2011), the best solution would be to perform both the tagging and parsing simultaneously. Check the tool citations for more information.

## 2. Notes on Methodology

### 2.1. On System Overview

Winge proposes a system based on common NLP modules: a tokenizer, a lemmatizer and a PoS tagger, in conjunction with the Morpheus morphological analyzer. The tagger can be trained on the LDT or PROIEL, but only the LDT can be used in the lemmatizer due to the incompatibility of PROIEL with the lemmas in Morpheus. 

The text is first tokenized according to the standard of the corpus, then the set of unique tokens is sent to Morpheus, which reports the possible analyses for each, which returns a tuple (morphological tag, lemma, word form with marked vowel lenghts). The tokenized text is fed to the PoS tagger and the lemmatizer. For each token, the predicted morphological tag and lemma are matched to one of the analyses given by Morpheus. 

 Important to note: Winge made several corrections to Morpheus' insufficient or wrong information when instances of such appeared. See [github](https://github.com/Alatius/morpheus/releases/tag/thesis).

### 2.2. On Tokenization

Enclictic particles (_-que_, _-ne_, _-ve_, etc.) are the biggest issue here. Winge handles these by not dividing the word if looks like a real Latin word undivised. The error this induces is irrelevant according to hsi stats.

### 2.3. On PoS Tagging 

_HunPos_, _RFTagger_ and the experimental _MATE_ tools were picked for PoS tagging. Sentences in the training set were randomly mixed, then partitioned into training, development, and test sets.

_will edit and finish this soon_
